### PR TITLE
feat: Add Avalanche support

### PR DESCRIPTION
**Pull request by cbachmeier**
## Description

_Linear ticket:_
_Slack thread:_
_Relevant docs:_

### Before
|    Mobile    |   Desktop    |
| ------------ | ------------ |
| paste_before | paste_before |

### After
|    Mobile    |   Desktop   |
| ------------ | ----------- |
| paste_after  | paste_after |

### Reproducing the error

1. 

### QA (ie manual testing)

- [ ] N/A

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

File: src/lib/state/multicall.tsx [hunk 1]
```diff
@@ -9,8 +9,15 @@ const multicall = createMulticall()
...
 
+/**
+ *
+ * @param chainId
+ * @returns The approximate number of blocks written to the corresponding chainId per Ethereum mainnet epoch.
+ */
 function getBlocksPerFetchForChainId(chainId: number | undefined): number {
   switch (chainId) {
+    case ChainId.AVALANCHE:
+      return 30
```
Comment (JFrankfurt) [inline src/lib/state/multicall.tsx:20]: @zzmp This aligns with my understanding of the intent of this function, can you confirm?

File: src/utils/getExplorerLink.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ const BLOCK_EXPLORER_PREFIXES: { [chainId: number]: string } = {
   [ChainId.CELO]: 'https://celoscan.io',
   [ChainId.CELO_ALFAJORES]: 'https://alfajores-blockscout.celo-testnet.org',
   [ChainId.BNB]: 'https://bscscan.com',
+  [ChainId.AVALANCHE]: 'https://snowtrace.io',
```
Comment (zzmp) [inline src/utils/getExplorerLink.ts:14]: I think they snagged the coolest name 😎

File: src/lib/state/multicall.tsx (hunk 1 above)
Comment (zzmp) [reply src/lib/state/multicall.tsx:20]: These numbers all seem out of date. Maybe they should be updated in a follow up? It should be appx. how many blocks there are in 12s.

File: src/lib/state/multicall.tsx [hunk 3]
```diff
@@ -9,8 +9,15 @@ const multicall = createMulticall()
 
 export default multicall
 
+/**
+ *
+ * @param chainId
+ * @returns The approximate number of blocks written to the corresponding chainId per Ethereum mainnet epoch.
```
Comment (zzmp) [inline src/lib/state/multicall.tsx:15]:
```suggestion
 * @returns The approximate whole number of blocks written to the corresponding chainId per Ethereum mainnet epoch.
```

File: src/lib/state/multicall.tsx (hunk 1 above)
Comment (zzmp) [reply src/lib/state/multicall.tsx:20]: I'm googling that it's about 1 block / 2 seconds, which would make this number 6. Is that wrong?

File: src/lib/hooks/useCurrencyLogoURIs.ts [hunk 4]
```diff
@@ -4,12 +4,13 @@ import { useMemo } from 'react'
...
 
 import EthereumLogo from '../../assets/images/ethereum-logo.png'
+import AvaxLogo from '../../assets/svg/avax_logo.svg'
 import BnbLogo from '../../assets/svg/bnb-logo.svg'
 import CeloLogo from '../../assets/svg/celo_logo.svg'
 import MaticLogo from '../../assets/svg/matic-token-icon.svg'
 import { isCelo, NATIVE_CHAIN_ID, nativeOnChain } from '../../constants/tokens'
 
-type Network = 'ethereum' | 'arbitrum' | 'optimism' | 'polygon' | 'smartchain' | 'celo'
+type Network = 'ethereum' | 'arbitrum' | 'optimism' | 'polygon' | 'smartchain' | 'celo' | 'avalanchec'
```
Comment (zzmp) [inline src/lib/hooks/useCurrencyLogoURIs.ts:13]:
```suggestion
type Network = 'ethereum' | 'arbitrum' | 'optimism' | 'polygon' | 'smartchain' | 'celo' | 'avalanche'
```

File: src/hooks/useStablecoinPrice.ts [hunk 5]
```diff
@@ -23,6 +24,7 @@ const STABLECOIN_AMOUNT_OUT: { [chainId: number]: CurrencyAmount<Token> } = {
   [ChainId.POLYGON]: CurrencyAmount.fromRawAmount(USDC_POLYGON, 10_000e6),
   [ChainId.CELO]: CurrencyAmount.fromRawAmount(CUSD_CELO, 10_000e18),
   [ChainId.BNB]: CurrencyAmount.fromRawAmount(USDT_BSC, 100e18),
+  [ChainId.AVALANCHE]: CurrencyAmount.fromRawAmount(USDT_AVALANCHE, 100e18),
```
Comment (zzmp) [inline src/hooks/useStablecoinPrice.ts:27]: Depending on volume we may want to go lower (as we did with Polygon).

File: src/graphql/data/util.tsx [hunk 6]
```diff
@@ -124,13 +125,15 @@ export const CHAIN_NAME_TO_CHAIN_ID: { [key in Chain]: ChainId } = {
...
+  // [Chain.Avalanche]: ChainId.AVALANCHE,
 }
 
 export function fromGraphQLChain(chain: Chain): ChainId {
   return CHAIN_NAME_TO_CHAIN_ID[chain]
 }
 
-export const BACKEND_CHAIN_NAMES: Chain[] = [Chain.Ethereum, Chain.Polygon, Chain.Optimism, Chain.Arbitrum, Chain.Celo]
+export const BACKEND_CHAIN_NAMES = [Chain.Ethereum, Chain.Polygon, Chain.Optimism, Chain.Arbitrum, Chain.Celo] as const
+export const BACKEND_UNSUPPORTED_CHAIN_IDS = [ChainId.BNB] as const
```
Comment (zzmp) [inline src/graphql/data/util.tsx:136]: nit: `BACKEND_NOT_YET_SUPPORTED_CHAINS`

File: src/graphql/data/util.tsx [hunk 7]
```diff
@@ -124,13 +125,15 @@ export const CHAIN_NAME_TO_CHAIN_ID: { [key in Chain]: ChainId } = {
...
   [Chain.Bnb]: ChainId.BNB,
+  // [Chain.Avalanche]: ChainId.AVALANCHE,
 }
 
 export function fromGraphQLChain(chain: Chain): ChainId {
   return CHAIN_NAME_TO_CHAIN_ID[chain]
 }
 
-export const BACKEND_CHAIN_NAMES: Chain[] = [Chain.Ethereum, Chain.Polygon, Chain.Optimism, Chain.Arbitrum, Chain.Celo]
+export const BACKEND_CHAIN_NAMES = [Chain.Ethereum, Chain.Polygon, Chain.Optimism, Chain.Arbitrum, Chain.Celo] as const
```
Comment (zzmp) [inline src/graphql/data/util.tsx:135]: nit: `BACKEND_SUPPORTED_CHAINS`

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/utils/getExplorerLink.ts (hunk 2 above)
Comment (cbachmeier) [reply src/utils/getExplorerLink.ts:14]: ❄️

File: src/lib/state/multicall.tsx (hunk 1 above)
Comment (cbachmeier) [reply src/lib/state/multicall.tsx:20]: @JFrankfurt do you remember the napkin math we did to find 30? Did we find that it was ~2.5tx/s?

File: src/lib/hooks/useCurrencyLogoURIs.ts (hunk 4 above)
Comment (cbachmeier) [reply src/lib/hooks/useCurrencyLogoURIs.ts:13]: It's actually supposed to be `avalanchec` for Avalanche C chain see:
https://github.com/Uniswap/assets/tree/master/blockchains/avalanchec

File: src/lib/state/multicall.tsx (hunk 1 above)
Comment (JFrankfurt) [reply src/lib/state/multicall.tsx:20]: we were estimating the number of blocks per ethereum block time, so roughly the number of blocks you'd expect avalanche to produce in 12s

Comment (JFrankfurt) [reply src/lib/state/multicall.tsx]: @cbachmeier after looking closer
1. block time on avalanche is kinda inconsistent. It looks like blocks are produced as work become available, not on a target time spacing.
2. I think I was looking at the network aggregate on avascan, not just the c-chain, when I came up with the initial estimate

Comment (cbachmeier) [reply src/lib/state/multicall.tsx]: @zzmp I've added a ticket to revisit the numbers

File: src/graphql/data/util.tsx [hunk 8]
```diff
@@ -147,8 +146,36 @@ export function isSupportedGQLChain(chain: Chain): chain is InterfaceGqlChain {
...
 
-export const BACKEND_CHAIN_NAMES = [Chain.Ethereum, Chain.Polygon, Chain.Optimism, Chain.Arbitrum, Chain.Celo] as const
-export const BACKEND_UNSUPPORTED_CHAIN_IDS = [ChainId.BNB] as const
+export function supportedChainIdFromGQLChain(chain: InterfaceGqlChain): ChainId
+export function supportedChainIdFromGQLChain(chain: Chain): ChainId | undefined
+export function supportedChainIdFromGQLChain(chain: Chain): ChainId | undefined {
+  return isSupportedGQLChain(chain) ? CHAIN_NAME_TO_CHAIN_ID[chain] : undefined
+}
+
+export function logSentryErrorForUnsupportedChain({
```
Comment (zzmp) [inline src/graphql/data/util.tsx]: Is this unused?

**Review by zzmp — APPROVED** [review]

File: src/graphql/data/util.tsx (hunk 8 above)
Comment (cbachmeier) [reply src/graphql/data/util.tsx]: nope it's used, it's a helper fn for Sentry errors relating to BE GQL providing data from an unsupported chain
Screenshot 2023-07-06 at 11 36 58 AM

**Pull request merged**